.github/workflows/codeql-analysis.yml

```
@@ -0,0 +1,44 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [master]
```

```suggestion
    branches: [main]
```
not sure if this essential but still..